samples/snippets/csharp/VS_Snippets_CLR/conceptual.disposable/cs/Foo.cs

```diff
@@ -9,5 +9,5 @@ public Foo()
         _bar = new Bar();
     }
 
-    public void Dispose() => _bar?.Dispose();
+    public void Dispose() => _bar.Dispose();
```

Comment:
@shadow-cs I'm confused about this check. Why would you remove the null-conditional operator without adding a null check?

Reply:
This is why I added the `TIP` in the MD, I found it confusing why the `null` check was there it sort of pointed out to a previously stated assumption that the GC may somehow put `null`s there (by reclaiming fields before the instance owning them) which is not the case. The `null`-check is only possible in this case if the `Bar` ctor throws which is not a pattern I'd generally recommend.

Reply:
My thought process is that it doesn't hurt to have it as a null-conditional as it will just no-op. 🤷🏼

Reply:
@IEvangelist I agree it doesn't hurt. But look at it from the other side, I began wondering why is the `null` check there, it shouldn't be necessary if `_bar` is initialized properly and then the `reclaimed/destroys` comes in and you begin wondering if it somehow possible `null` will ever be there? Is it possible the runtime does this? In fact, it doesn't which is sort of the point I'm trying to make. But maybe I'm just overthinking it. But chances are, there are other over-thinkers like me out there 😉